### PR TITLE
Adapt projects to not refer to embedded apps directory.

### PR DESCRIPTION
**Pull request by ksubrama**

### CODE REVIEW

**Comment by ksubrama** [comment]
Once this is ready to be merged, I will rebase against master and bump the Gemfile.lock against the real commits on master.

File: config/projects/angrychef.rb [hunk 1]
```diff
@@ -47,6 +47,17 @@
 compress :dmg
 
 package :msi do
+  install_path = Pathname.new(install_dir)
+
+  # Find path in which the Chef gem is installed
+  search_pattern = "#{install_path}/**/gems/chef-[0-9]*-mingw32/ext/win32-eventlog/chef-log.dll"
```
Comment (chefsalim) [inline config/projects/angrychef.rb]: Will this work for 64-bit also?

File: config/projects/angrychef.rb [hunk 2]
```diff
@@ -47,6 +47,17 @@
...
+  dll_path  = Pathname.glob(search_pattern).find { |path| path.file? }
```
Comment (chefsalim) [inline config/projects/angrychef.rb]: Wrap in a utility function to re-use the code in multiple places?

File: jenkins/verify-chef.sh [hunk 3]
```diff
@@ -74,6 +74,12 @@ export FORCE_FFI_YAJL
...
-# Test against the appbundle'd Chef
-cd /opt/$PROJECT_NAME/embedded/apps/chef
+# Test against the vendored Chef gem
+cd /opt/$PROJECT_NAME/embedded/lib/ruby/gems/*/gems/chef-[0-9]*
+
+if [ ! -f "Gemfile.lock" ]; then
+  echo "Chef gem does not contain a Gemfile.lock! This is needed to run any tests."
+  exit 1
+fi
+
```
Comment (schisamo) [inline jenkins/verify-chef.sh]: We'll need to ensure these changes work on all *nix platforms...worried about more esoteric things like FreeBSD and Solaris. A simple test build should verify things.

File: resources/angrychef/msi/source.wxs.erb [hunk 4]
```diff
@@ -47,11 +47,6 @@
               </Component>
             </Directory>
             <Directory Id="PSMODULES" Name="modules" >
-              <Directory Id="PSMODULESCHEF" Name="chef" >
-                <Component Id="ChefPSModuleFile" Guid="{DF0E99BF-1877-4099-85A4-067C84A743F2}" >
-                  <File Id="chef.psm1" Source="$(var.ProjectSourceDir)\embedded\apps\chef\distro\powershell\chef\chef.psm1" KeyPath="yes" />
-                </Component>
-              </Directory>
```
Comment (schisamo) [inline resources/angrychef/msi/source.wxs.erb]: Why are these changes part of this PR? They appear unrelated.

File: config/projects/angrychef.rb [hunk 5]
```diff
@@ -47,6 +47,17 @@
...
+
+  raise "Could not find the log dll at `#{search_pattern}'!" if dll_path.nil?
+  dll_relative_path = windows_safe_path(dll_path.relative_path_from(install_path).to_s)
+
   upgrade_code "D7FDDC1A-7668-404E-AD2F-61F875632A9C"
+  wix_candle_extension 'WixUtilExtension'
```
Comment (schisamo) [inline config/projects/angrychef.rb]: This change is already part of https://github.com/chef/omnibus-chef/pull/531.

File: config/projects/angrychef.rb (hunk 1 above)
Comment (ksubrama) [inline config/projects/angrychef.rb]: No better or worse than what it currently does.  The paths are still going to contain mingw32/ext/win32-eventlog unless we rename it.  Whether or not the dll we compile will actually work...  *shrug*.  We already use an old deprecated API IIRC - so we'll probably need to take a look at it separately when dealing with Nano.

**Comment by schisamo** [comment]
Overall it feels like all the code introduced in the `package :msi` block should be some sort of helper exposed in `Omnibus::Packager::MSI` (like `Omnibus::Packager::MSI#wix_light_extension`, `Omnibus::Packager::MSI#wix_candle_extension`). That would remove the duplication between the Chef and ChefDK project definitions.

File: config/projects/angrychef.rb (hunk 2 above)
Comment (ksubrama) [inline config/projects/angrychef.rb]: There wasn't an easy place to refactor it :(  I could put it in the MSI packager but that seems overly specific.  Omnibus util seems like a good place but even there, the particular strategy of using a random path glob/search pattern seemed sub-optimal.  What would really be nice is if I could interrogate the software definition itself to see the list of gems it installed or being able to run "gem which".  I do have an item to refactor this code/handle it better - for now, it follows the current workaround pattern in push-jobs.

File: resources/angrychef/msi/source.wxs.erb (hunk 4 above)
Comment (ksubrama) [inline resources/angrychef/msi/source.wxs.erb]: There is no embedded apps directory any more :).  I moved the responsibility of deploying these files into the software definition of chef.

File: config/projects/angrychef.rb (hunk 5 above)
Comment (ksubrama) [inline config/projects/angrychef.rb]: Thank you!

File: config/projects/angrychef.rb (hunk 2 above)
Comment (ksubrama) [inline config/projects/angrychef.rb]: And on second thoughts - schisamo did point out that we already have some fairly specific helper methods in the MSI packager.  I'll refactor this (and push-jobs).

File: jenkins/verify-chef.sh (hunk 3 above)
Comment (ksubrama) [inline jenkins/verify-chef.sh]: http://manhattan.ci.chef.co/job/chef-12-test/418/  It looks good!
